lib/saluki-app/src/logging.rs

```diff
@@ -115,19 +133,47 @@ fn initialize_logging_inner(
             .replace(LoggingAPIHandler::new(shared_level_filter.clone(), reload_handle));
     }
 
+    let adp_log_file = std::env::var("DD_ADP_LOG_FILE").unwrap_or(DEFAULT_ADP_LOG_FILE.to_string());
+    let log_file_max_size = std::env::var("DD_LOG_FILE_MAX_SIZE")
+        .map(|s| s.parse::<u64>().unwrap_or(DEFAULT_LOG_FILE_MAX_SIZE))
+        .unwrap_or(DEFAULT_LOG_FILE_MAX_SIZE);
+    let log_file_max_rolls = std::env::var("DD_LOG_FILE_MAX_ROLLS")
+        .map(|s| s.parse::<usize>().unwrap_or(DEFAULT_LOG_FILE_MAX_ROLLS))
+        .unwrap_or(DEFAULT_LOG_FILE_MAX_ROLLS);
+    let file_appender = BasicRollingFileAppender::new(
+        adp_log_file,
+        RollingConditionBasic::new().max_size(log_file_max_size),
+        log_file_max_rolls,
+    )?;
+    let (file_nb, guard) = tracing_appender::non_blocking(file_appender);
+    let file_level_filter = EnvFilter::builder()
```

Comment:
Cleaner to just take advantage of `shared_level_filter`, I think, since it already does this much for us, minus any ability to be dynamically updated.

Reply:
Had type issues substituting `shared_level_filter` for `file_level_filter` here as well.